src/bin/sccache-dist/main.rs

```diff
@@ -556,7 +565,14 @@ impl SchedulerIncoming for Scheduler {
                         "Non-terminated job {} was cleaned up in server pruning",
                         job_id
                     );
-                    assert!(jobs.remove(&job_id).is_some())
+                    // A job may be missing here if it failed to allocate
+                    // initially, so just warn it's not present.
+                    if !jobs.remove(&job_id).is_some() {
```

Comment:
Nit: Maybe this is better expressed as the positive `if jobs.remove(...).is_none() {`, which is a little more in line with the comment above?